viewer/db.js

```diff
@@ -987,31 +1002,208 @@ exports.getHunt = async (id) => {
 };
 
 // Shortcut DB interactions
+// since 3.0 shortcuts are synced across clusters if user's elasticsearch is configured
+// determine if shorcuts have been synced from the local db to the remote db (remote db = user's db)
+// if there are shortcuts to sync, adds them to the remote db and deduplicates
+// shortcut names by appending the cluster or index name to the shortcut name
+async function initialShortcutsSync () {
+  if (internals.multiES) { return; } // don't sync shortcuts for multies
+
+  const { body: doc } = await internals.usersClient7.indices.getMapping({
+    index: internals.remoteShortcutsIndex
+  });
+
+  let initSync;
+  // eslint-disable-next-line no-unreachable-loop
+  for (const index in doc) {
+    initSync = doc[index]?.mappings?._meta?.initSync;
+    break;
+  }
+
+  if (initSync) { return; } // already been synced, don't need to do anything
+
+  const msg = `initial sync from local shortcuts index (${internals.localShortcutsIndex}) to remote index (${internals.remoteShortcutsIndex})`;
+
+  console.log(msg);
+
+  try {
+    // fetch shortcuts from remote and local indexes
+    const [{ body: remoteResults }, { body: localResults }] = await Promise.all([
+      exports.searchShortcuts({}), exports.searchShortcutsLocal({})
+    ]);
+
+    if (!localResults.hits.hits.length) { return; } // nothing to sync to remote db
+
+    // add each of the local shortcuts to the remote shortcuts db (remote db = user's es)
+    // but first, compare the local shortcuts to the remote shortcuts to determine
+    // if any shortcuts have the same name in the remote db (we can assume that
+    // none of the local shortcuts will be the same as the remote db shortcuts
+    // since they have not been synced before)
+    for (const localShortcut of localResults.hits.hits) {
+      for (const remoteShortcut of remoteResults.hits.hits) {
+        if (remoteShortcut._source.name === localShortcut._source.name) {
```

Comment:
Would be cool if you could compare the short cut and only do this if they are different values. If the same values keep the remote and delete the local one and wait for remote to be copied back down?